js/containers/feedback-panel.js

```diff
@@ -0,0 +1,234 @@
+    this.studentRowRef = this.studentRowRef.bind(this)
+  }
+
+
+  makeOnlyNeedReview() {
+    this.setState({showOnlyNeedReview: true})
+  }
+
+  makeShowAll() {
+    this.setState({showOnlyNeedReview: false})
+  }
+
+  showFeedback() {
+    this.setState({
+      showFeedbackPanel: true
+    })
+  }
+
+  hideFeedback() {
+    this.setState({
+      showFeedbackPanel: false
+    })
+  }
+
+  answerIsMarkedComplete(answer) {
+    const feedbackId = answer.get('feedbacks') && answer.get('feedbacks').last()
+    const feedback = this.props.feedbacks.get(feedbackId)
+    return feedback && feedback.get("hasBeenReviewed")
+  }
+
+  enableText(event) {
+    this.props.enableFeedback(this.props.question.get('key'), {feedbackEnabled: event.target.checked})
+  }
+
+  enableScore(event) {
+    this.props.enableFeedback(this.props.question.get('key'), {scoreEnabled: event.target.checked})
+  }
+
+  setMaxScore(event) {
+    const value = parseInt(event.target.value) || null
+    this.props.enableFeedback(this.props.question.get('key'), {maxScore: value})
+  }
+
+  studentRowRef(index){
+    return `student-row-${index}`
+  }
+
+  scrollStudentIntoView(eventProxy) {
+    const index = eventProxy.target.value
+    const ref = this.studentRowRef(index)
+    const itemComponent = this.refs[ref]
+    if (itemComponent) {
+      const domNode = ReactDom.findDOMNode(itemComponent)
+      domNode.scrollIntoView()
+    }
+  }
+
+  renderGettingStarted() {
+    return(
+      <div className="gettingStarted">
+        <div className="explainer">
+          To start, choose the type of feedback you want to leave in the Feedback Type settings above.
+        </div>
+        <div className="arrow">⤴</div>
+      </div>
+    )
+  }
+
+  render() {
+    const question      = this.props.question
+    const showing       = this.state.showFeedbackPanel
+    const prompt        = question.get('prompt')
+    const number        = question.get('questionNumber')
+    const answers       = this.props.answers
+
+    const realAnswers     = answers.filter(a => a.get('type') != 'NoAnswer')
+    const needingFeedback = realAnswers.filter( a => ! this.answerIsMarkedComplete(a) )
+
+    const filteredAnswers = this.state.showOnlyNeedReview ? needingFeedback : answers
+    const numAnswers       = realAnswers.count()
+    const numNoAnswers     = answers.count() - realAnswers.count()
+    let   numNeedsFeedback = needingFeedback.count()
+    let   numFeedbackGiven = numAnswers - numNeedsFeedback
+
+    const scoreEnabled = question.get('scoreEnabled')
+    const feedbackEnabled = question.get('feedbackEnabled')
+    const maxScore = question.get('maxScore')
+
+    const showGettingStarted = (!scoreEnabled) && (!feedbackEnabled)
+    const studentsPulldown = filteredAnswers.map( (a) => {
+      return {
+        realName: a.getIn(['student', 'realName']),
+        id: a.getIn(['student','id']),
+        answer: a,
+      }
+    })
```

Comment:
FYI: I think arrow functions that only return something can be written without return statement and { }, in this case:
```
.map(a => {realName: ..., id: ..., answer: ...}
```